airflow/providers/apache/kafka/hooks/consumer.py

```diff
@@ -0,0 +1,60 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+from typing import Any, Sequence
+
+from confluent_kafka import Consumer
+
+from airflow.exceptions import AirflowException
+from airflow.hooks.base import BaseHook
+
+
+class KafkaConsumerHook(BaseHook):
+    """KafkaConsumerHook
+
+    A hook for creating a Kafka Consumer
+    :param config: A config dictionary to use with confluent_kafka library, defaults to None
+    """
+
+
+    def __init__(
+        self,
+        topics: Sequence[str],
+        config: dict[Any, Any] | None = None,
+    ) -> None:
+        super().__init__()
+
+        self.config: dict[Any, Any] = config or {}
+        self.topics = topics
+
+        if not self.config.get("group.id", None):
+            raise AirflowException(
+                "The 'group.id' parameter must be set in the config dictionary'. Got <None>"
+            )
+
+        if not self.config.get("bootstrap.servers", None):
+            raise AirflowException("config['bootsrap.servers'] must be provided.")
+
+    def get_consumer(self) -> Consumer:
```

Comment:
WDYT about splitting this method into two:
1. `get_consumer()` which retrieves the `Consumer` object
2. `subscribe_to_topics()` which uses the `Consumer` object and subscribes to an input set of topics?

The subscribe functionality could be used atomically if folks wanted to use it and also use the `Consumer` object via `get_consumer()` for other actions which might not have an explicit hook method in this provider.

Reply:
Consumers that aren't subscribed to topics are a no-op so I don't think there's really a reason to do that.